packages/block-library/src/block/edit.js

```diff
@@ -147,10 +147,13 @@ function getContentValuesFromInnerBlocks( blocks, defaultValues ) {
 				defaultValues[ blockId ][ attributeKey ]
 			) {
 				content[ blockId ] ??= { values: {} };
-				// TODO: We need a way to represent `undefined` in the serialized overrides.
-				// Also see: https://github.com/WordPress/gutenberg/pull/57249#discussion_r1452987871
 				content[ blockId ].values[ attributeKey ] =
-					block.attributes[ attributeKey ];
+					block.attributes[ attributeKey ] === undefined
```

Comment:
It makes me a little nervous that this might have unintended consequences on other blocks, but it's not a big issue right now while there's a limited number of blocks supported.

An alternative could be maintaining a list of blocks and attributes near this code that this fix/hack should apply to.

Reply:
I think we can (and probably should) adjust this as we support more blocks. There's already a TODO comment below this line so I think it's enough for now?